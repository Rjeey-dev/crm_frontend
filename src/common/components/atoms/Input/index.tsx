import React from 'react';

import Select from "common/components/atoms/Select/index";

const Input = ({ ...props }) => {
    if (props.type === 'textarea') {
        return <textarea {...props}/>
    } else if (props.type === 'select') {
        return <Select {...props} >{props.children}</Select>
    }

    return <input {...props}/>
};

export default Input;