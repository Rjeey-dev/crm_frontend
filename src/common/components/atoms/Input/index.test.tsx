import { shallow } from 'enzyme';
import React from 'react';

import Input from './index';

it("Renders <Input/> textarea", () => {
    const output = shallow(
        <Input classes='class' type='textarea'/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Input/> select", () => {
    const output = shallow(
        <Input classes='class' type='select'/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Input/> text", () => {
    const output = shallow(
        <Input classes='class'/>
    ).html();

    expect(output).toMatchSnapshot();
});