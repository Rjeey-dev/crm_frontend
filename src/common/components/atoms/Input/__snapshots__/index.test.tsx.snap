// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Input/> select 1`] = `"<select classes=\\"class\\" type=\\"select\\"></select>"`;

exports[`Renders <Input/> text 1`] = `"<input classes=\\"class\\"/>"`;

exports[`Renders <Input/> textarea 1`] = `"<textarea classes=\\"class\\" type=\\"textarea\\"></textarea>"`;
